src/coreclr/tools/Common/TypeSystem/Canon/TypeSystemContext.Canon.cs

```diff
@@ -92,6 +92,7 @@ protected internal virtual TypeDesc ConvertToCanon(TypeDesc typeToConvert, Canon
             throw new NotSupportedException();
         }
 
+        internal partial void InternalGetSupportsUniversalCanon(ref bool flag) => flag = SupportsUniversalCanon;
```

Comment:
Why do we need this ugly hack?

Reply:
Because Canon support is in partial files, we don't force canon support on every user of the type system, and ILVerify doesn't include it.

Reply:
Would it better to fix this by moving the `ComputeInstanceFieldLayout` into the concrete type systems and leave it abstract in the base one?

Reply:
I thought about it, but that would basically mean that the change we're doing now (changing qualification for Sequential layout) would not be automatically reflected in unit tests because tests don't use the compiler type system context. So the logic would not be exercised at all unless we don't forget to keep it in sync.

Reply:
I think that it is ok. There is no one right way to do field layout, so it makes sense to allow it to be configurable and unit tests testing a subset of the possible configurations.

Reply:
Ok, in that case I'll redo this on the NativeAOT side. Test changes won't be needed because we can just keep using the old logic.